vscode/src/completions/vscodeInlineCompletionItemProvider.ts

```diff
@@ -161,86 +165,96 @@ export class InlineCompletionItemProvider implements vscode.InlineCompletionItem
+                              lastTriggerPosition: position,
+                              lastTriggerCurrentLinePrefix: document.lineAt(position).text.slice(0, position.character),
+                              lastTriggerNextNonEmptyLine: getNextNonEmptyLine(
+                                  document.getText(
+                                      new vscode.Range(position, document.lineAt(document.lineCount - 1).range.end)
+                                  )
+                              ),
+                              lastTriggerSelectedInfoItem: context?.selectedCompletionInfo?.text,
+                              result: {
+                                  logId: result.logId,
+                                  items: result.items,
+                              },
+                          }
+                        : undefined
+            }
 
-        // A completion that won't be visible in VS Code will not be returned and not be logged.
-        if (
-            !isCompletionVisible(
-                items,
+            const items = this.processInlineCompletionsForVSCode(
+                result.logId,
                 document,
-                docContext,
-                context,
-                this.config.completeSuggestWidgetSelection,
-                abortController.signal
+                position,
+                result.items,
+                context
             )
-        ) {
-            return null
-        }
 
-        const event = CompletionLogger.completionEvent(result.logId)
-        if (items.length > 0) {
-            CompletionLogger.suggested(result.logId, InlineCompletionsResultSource[result.source], items[0] as any)
-        } else {
-            CompletionLogger.noResponse(result.logId)
-        }
+            // A completion that won't be visible in VS Code will not be returned and not be logged.
+            if (
+                !isCompletionVisible(
+                    items,
+                    document,
+                    docContext,
+                    context,
+                    this.config.completeSuggestWidgetSelection,
+                    abortController.signal
+                )
+            ) {
+                return null
+            }
 
-        const completionResult: vscode.InlineCompletionList = { items }
+            const event = CompletionLogger.completionEvent(result.logId)
+            if (items.length > 0) {
+                CompletionLogger.suggested(result.logId, InlineCompletionsResultSource[result.source], items[0] as any)
+            } else {
+                CompletionLogger.noResponse(result.logId)
+            }
+
+            const completionResult: vscode.InlineCompletionList = { items }
 
-        ;(completionResult as any).completionEvent = event
+            ;(completionResult as any).completionEvent = event
 
-        return completionResult
+            return completionResult
+        } catch (error: unknown) {
```

Comment:
We can omit the `unknown` type here because we have [the `strict ` mode](https://www.typescriptlang.org/tsconfig#useUnknownInCatchVariables) enabled:

```suggestion
        } catch (error) {
```